src/libtiled/maptovariantconverter.cpp

```diff
@@ -263,15 +270,41 @@ QVariant MapToVariantConverter::toVariant(const Tileset &tileset,
     return tilesetVariant;
 }
 
-QVariant MapToVariantConverter::toVariant(const TemplateGroup &templateGroup, int firstTid) const
+QVariant MapToVariantConverter::toVariant(const TemplateGroup &templateGroup, int firstTid)
 {
     QVariantMap templateGroupVariant;
 
-    templateGroupVariant[QLatin1String("firsttid")] = firstTid;
+    if (firstTid > 0) {
```

Comment:
Since you have essentially two entirely different implementations based on whether a `firstTid` is passed or not, I think it would be better to have two separate overloads for this. In fact the version that doesn't take a `firstTid` can be inlined in `MapToVariantConverter::toVariant(const TemplateGroup &templateGroup, QDir &directory)`.

This is different for the function for tilesets, since those can be embedded.